src/components/common/InfoCard/index.jsx

```diff
@@ -37,50 +37,51 @@ export const InfoCard = (props) => {
           )}
         </div>
 
-        {Object.entries(props?.data).map(([key, value]) => {
-          if (Array.isArray(value)) {
-            return (
-              <InfoCard collapse={true} type={key} data={value}>
-                {value.map((item, index) => (
-                  <InfoCard
-                    collapse={true}
-                    key={Math.random()}
-                    type={(index + 1).toString()}
-                    data={item}
-                  />
-                ))}
-              </InfoCard>
-            );
-          } else if (typeof value === 'object') {
-            return (
-              <InfoCard
-                key={Math.random()}
-                collapse={true}
-                type={isNaN(key) ? key : (parseInt(key) + 1).toString()}
-                data={value}
-              />
-            );
-          }
+        {props.data &&
+          Object.entries(props.data).map(([key, value]) => {
+            if (Array.isArray(value)) {
+              return (
+                <InfoCard
+                  key={`${key}-${value}`}
+                  collapse={true}
+                  type={key}
+                  data={value}>
+                  {value.map((item, index) => (
+                    <InfoCard
+                      collapse={true}
+                      key={index}
+                      type={(index + 1).toString()}
+                      data={item}
+                    />
+                  ))}
+                </InfoCard>
+              );
+            } else if (typeof value === 'object') {
+              return (
+                <InfoCard
+                  key={Math.random()}
+                  collapse={true}
+                  type={isNaN(key) ? key : (parseInt(key) + 1).toString()}
+                  data={value}
+                />
+              );
+            }
 
-          return <InfoRow key={Math.random()} label={key} value={value} />;
-        })}
+            return <InfoRow key={Math.random()} label={key} value={value} />;
+          })}
 
         {/* contracts inside the transaction */}
         <div style={{ display: 'none', flexDirection: 'row', gap: '1rem' }}>
           <div>
             {/* info cards of contracts */}
             {props.data?.contracts?.map((contract, index) => (
-              <>
+              <React.Fragment key={`contract-${index}`}>
```

Comment:
If recommend use `Fragment` rather `React.Fragment` since no other place we are using it